src/display/base.jl

```diff
@@ -102,15 +102,17 @@ import Base.Docs: doc
 
 isanon(f) = startswith(string(typeof(f).name.mt.name), "#")
 
-@render Inline f::Function begin
-  name = repr(f)
-  name_without_module = string(f)
-  binding = Docs.Binding(typeof(f).name.module, Symbol(name_without_module))
+@render i::Inline f::Function begin
   if isanon(f)
-    span(".syntax--support.syntax--function", "λ")
+    sig, link = view(first(methods(f)))
+    r(x) = render(i, x)
+    defaultrepr(f, true, span(".syntax--support.syntax--function",
+                              c("function ", r(sig), " at ", r(link))))
```

Comment:
Technically, anonymous functions can have multiple methods, so maybe we should just use the normal rendering code here with a `λ` as the function name?

Reply:
I think the link to the definition line is pretty useful, and having the function signature is nice too. And anonymous function would almost always have exactly one method. Maybe add the methods as a subtree when there is more than one?

Reply:
What do you think about something like
```
@render Inline f::Function begin
  if isanon(f)
    LazyTree(
      span(".syntax--support.syntax--function", "λ"),
      ()->methods(f)
    )
  else
    name = repr(f)
    name_without_module = string(f)
    binding = Docs.Binding(typeof(f).name.module, Symbol(name_without_module))
    LazyTree(
      span(".syntax--support.syntax--function", name),
      ()->[(CodeTools.hasdoc(binding) ? [md_hlines(doc(binding))] : [])..., methods(f)]
    )
  end
end
```

That renders as 
![image](https://user-images.githubusercontent.com/6735977/87325497-d07ace80-c531-11ea-9fb6-16d265a10030.png)

Reply:
That would defeat the purpose of the PR, because it doesn't show the function's fields (variables closed over)

Reply:
I didn't actually realize that happened because I only tried non-closue anon functions :)

I think I'd still prefer a method list display as well though.